main.go

```diff
@@ -60,6 +60,14 @@ func main() {
 	}
 	v = append(v, fmt.Sprintf("spec: %s", specs.Version))
 	app.Version = strings.Join(v, "\n")
+
+	root := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
I've changed it to be used only when `geteuid() !=0`.  I think a shared directory for all users is a very uncommon use case, while most users would probably prefer to have runc working out of the box and behave like other applications that use `XDG_RUNTIME_DIR`.  The first case is still possible though by overriding the root directory, and I think it is preferable to require extra configuration for an uncommon use like a shared directory for all users.